lsp-ui-sideline.el

```diff
@@ -404,114 +418,129 @@ Push sideline overlays on `lsp-ui-sideline--ovs'."
+           (inhibit-modification-hooks t)
+           symbols)
+      (setq lsp-ui-sideline--tag tag
+            lsp-ui-sideline--last-width (window-text-width))
+      (when (and line-changed lsp-ui-sideline-show-diagnostics)
+        (lsp-ui-sideline--diagnostics bol eol))
+      (when (and lsp-ui-sideline-show-code-actions
+                 (or (lsp--capability "codeActionProvider")
+                     (lsp--registered-capability "textDocument/codeAction")))
+        (lsp-request-async
+         "textDocument/codeAction"
+         (if (eq lsp-ui-sideline-update-mode 'line)
+             (list :textDocument doc-id
+                   :range (lsp--region-to-range bol eol)
+                   :context (list :diagnostics (lsp-cur-line-diagnostics)))
+           (lsp--text-document-code-action-params))
+         (lambda (actions) (lsp-ui-sideline--code-actions actions bol eol))
+         :mode 'tick
+         :cancel-token :lsp-ui-code-actions))
+      ;; Go through all symbols and request hover information.  Note that the symbols are
+      ;; traversed backwards as `forward-symbol' with a positive argument will jump just past the
+      ;; current symbol.  By going from the end of the line towards the front, point will be placed
+      ;; at the beginning of each symbol.  As the requests are first collected in a list before
+      ;; being processed they are still sent in order from left to right.
+      (when (and lsp-ui-sideline-show-hover (or line-changed line-modified) (lsp--capability "hoverProvider"))
+        (setq lsp-ui-sideline--last-tick-info this-tick
+              lsp-ui-sideline--previous-line this-line)
+        (save-excursion
+          (goto-char eol)
+          (while (and (> (point) bol)
+                      (progn (forward-symbol -1)
+                             (>= (point) bol)))
+            (let* ((symbol (thing-at-point 'symbol t))
+                   (bounds (bounds-of-thing-at-point 'symbol))
+                   (parsing-state (syntax-ppss))
+                   (in-string (nth 3 parsing-state))
+                   (outside-comment (eq (nth 4 parsing-state) nil)))
+              ;; Skip strings and comments
+              (when (and symbol (not in-string) outside-comment)
+                (push (list symbol bounds (list :line (1- line-widen) :character (- (point) bol))) symbols))))
+          (let ((last-symbol (1- (length symbols)))
+                list-infos)
+            (--each-indexed symbols
+              (-let (((symbol bounds position) it)
+                     (index it-index))
+                (lsp-request-async
+                 "textDocument/hover"
+                 (lsp-make-hover-params :text-document doc-id :position position)
+                 (lambda (info)
+                   (and info (push (list symbol bounds info) list-infos))
+                   (when (= index last-symbol)
+                     (lsp-ui-sideline--display-all-info list-infos tag bol eol)))
```

Comment:
This code assumes that the server will always respond, even if it's an empty response.
It also assumes that `lsp-request-async` will always call the callback, even on empty response

Reply:
1. it won't be called in case of an error. In this case, `:error-handler` will be called.
2. In won't be called if the buffer is edited(this is what `:mode 'tick` is for). In this case `:cancel-handler` will be called.

Reply:
Thanks